internal/services/compute/orchestrated_virtual_machine_scale_set_resource.go

```diff
@@ -1160,6 +1212,28 @@ func resourceOrchestratedVirtualMachineScaleSetUpdate(d *pluginsdk.ResourceData,
 		update.Zones = pointer.To(zones.ExpandUntyped(d.Get("zones").(*schema.Set).List()))
 	}
 
+	if d.HasChange("rolling_upgrade_policy") {
+		upgradePolicy := virtualmachinescalesets.UpgradePolicy{}
+		if existing.Model.Properties.UpgradePolicy == nil {
+			upgradePolicy = virtualmachinescalesets.UpgradePolicy{
+				Mode: pointer.To(virtualmachinescalesets.UpgradeMode(d.Get("upgrade_mode").(string))),
+			}
+		} else {
+			upgradePolicy = *existing.Model.Properties.UpgradePolicy
+			upgradePolicy.Mode = pointer.To(virtualmachinescalesets.UpgradeMode(d.Get("upgrade_mode").(string)))
+		}
+
+		rollingRaw := d.Get("rolling_upgrade_policy").([]interface{})
+		zones := zones.ExpandUntyped(d.Get("zones").(*schema.Set).List())
+		rollingUpgradePolicy, err := ExpandVirtualMachineScaleSetRollingUpgradePolicy(rollingRaw, len(zones) > 0, false)
+		if err != nil {
+			return err
```

Comment:
could we wrap this error message?
```suggestion
			return fmt.Errorf("expanding `rolling_upgrade_policy`: %+v", err)
```

Reply:
updated. Thanks.